### PR TITLE
feat: Implement fixed vec arbitrary manually for NewPooledTransactions68

### LINKED ISSUE #6173: Implement arbitrary manually for NewPooledTransactions68
### Describe the feature

with #6090 rlp decoding of the type enforces same length of the internal Vecs

The proptest strategy generates a fixed size of all vecs. It would be nice if the arbitrary implementation of this types generates a random fixed size instead

https://github.com/paradigmxyz/reth/blob/2b97e0cb9fa2b09ced58488c3dc2c9a79ad32bf9/crates/net/eth-wire/src/types/broadcast.rs#L262

## TODO

implement arbitrary manually for the type.
Perhaps the easiest approach would be to expand the macro, copy and modify the generated code

### Additional context

_No response_

### LINKED ISSUE #6202: arbitrary impl `NewPooledTransactionHashes`: choose sizes from limited range
### Describe the feature

building onto https://github.com/paradigmxyz/reth/issues/6173, it would be helpful if the values generated for `sizes`, are limited to the range `0..0x20000`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by supernovahs**
closes #6173 
closes #6202

### CODE REVIEW

File: crates/net/eth-wire/src/types/broadcast.rs [hunk 1]
```diff
@@ -283,25 +288,32 @@ pub struct NewPooledTransactionHashes68 {
...
+    type Parameters = ();
+
+    fn arbitrary_with(_args: ()) -> Self::Strategy {
+        (any::<usize>(), any::<u8>(), any::<usize>(), any::<B256>())
+            .prop_map(|(size, type_val, size_val, hashes_val)| NewPooledTransactionHashes68 {
+                types: vec![type_val; size],
+                sizes: vec![size_val; size],
+                hashes: vec![hashes_val; size],
+            })
+            .boxed()
```
Comment (mattsse) [inline crates/net/eth-wire/src/types/broadcast.rs:313]: I'd like to changes this a bit, now that we have this manually

types should be restricted to the TxType values, possible tx types, and ideally randomized vecs

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, only two nits

**Comment by emhane** [comment]
lgtm, not familiar with `arbitrary` though so waiting on @mattsse to see if anything can be simplified

File: crates/net/eth-wire/src/types/broadcast.rs [hunk 2]
```diff
@@ -304,16 +304,37 @@ impl Arbitrary for NewPooledTransactionHashes68 {
...
+    // Example conversion logic
+    match tx_type {
+        TxType::Legacy => 1,
+        TxType::EIP2930 => 2,
+        TxType::EIP1559 => 3,
+        TxType::EIP4844 => 4,
+        #[cfg(feature = "optimism")]
+        TxType::DEPOSIT => 5,
+    }
+}
```
Comment (mattsse) [inline crates/net/eth-wire/src/types/broadcast.rs]: I don't think we need this `TxType as u8` should do it

File: crates/net/eth-wire/src/types/broadcast.rs [hunk 3]
```diff
@@ -304,16 +304,37 @@ impl Arbitrary for NewPooledTransactionHashes68 {
...
     fn arbitrary_with(_args: ()) -> Self::Strategy {
-        (any::<usize>(), any::<u8>(), any::<usize>(), any::<B256>())
-            .prop_map(|(size, type_val, size_val, hashes_val)| NewPooledTransactionHashes68 {
-                types: vec![type_val; size],
-                sizes: vec![size_val; size],
-                hashes: vec![hashes_val; size],
+        (vec(any::<TxType>(), 1..100), vec(any::<usize>(), 1..100), vec(any::<B256>(), 1..100))
+            .prop_map(|(types, sizes, hashes)| {
+                // Ensure the vectors have the same length
+                let min_length = *[types.len(), sizes.len(), hashes.len()].iter().min().unwrap();
```
Comment (mattsse) [inline crates/net/eth-wire/src/types/broadcast.rs:310]: I wonder if this can improved by using a random length value for all the vec strategies instead?

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**